### ISSUE TITLE
Sphinx 2.0.0  'NoneType' object has no attribute 'groupdict' with napoleon

### ISSUE THREAD

**Issue by alertedsnake** [sphinx-doc/sphinx#6225]
**Describe the bug**
Exception raised when trying to use napoleon with a docstring like this:
```
Raises:
    :py:class:`MyException`
```

**To Reproduce**
Run `sphinx-build docs` using `sphinx.ext.napoleon` and `'sphinx.ext.autodoc', using default settings.

```
$ sphinx-build . out
Running Sphinx v2.0.0
loading intersphinx inventory from http://docs.python.org/3.6/objects.inv...
intersphinx inventory has moved: http://docs.python.org/3.6/objects.inv -> https://docs.python.org/3.6/objects.inv
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 2 source files that are out of date
updating environment: 2 added, 0 changed, 0 removed
reading sources... [ 50%] out/foo                                                                                            
Exception occurred:
  File "/home/michael/test/.venv/lib/python3.6/site-packages/sphinx/ext/napoleon/docstring.py", line 703, in _parse_raises_section
    m = self._name_rgx.match(_type).groupdict()
AttributeError: 'NoneType' object has no attribute 'groupdict'
The full traceback has been saved in /tmp/sphinx-err-gza3ubmq.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
```

**Expected behavior**
No exception

**Your project**
[test.zip](https://github.com/sphinx-doc/sphinx/files/3024482/test.zip)

**Environment info**
- OS: Linux
- Python version: 3.6.8
- Sphinx version: 2.0.0
- Sphinx extensions:  sphinx.ext.autodoc, sphinx.ext.napoleon
- Extra tools:

**Comment by tk0miya** [comment]
Thank you for reporting. It seems this would be related with #6220.

**Comment by alertedsnake** [comment]
@tk0miya yup!  Saw that one, but it's a different error so I wasn't sure if it was the same thing or just tangentially related.

**Comment by tk0miya** [comment]
Fixed by #6237.
Thank you for reporting!
